.github/actions/setup-build/action.yml

```diff
@@ -0,0 +1,226 @@
+            exit 1
+          }
+          if ($ParsedWinMsvcVersion.Revision -ne -1) {
+            Write-Output "::error::Unsupported Windows MSVC version (revision version was specified): `"${WinMsvcVersion}`"."
+            exit 1
+          }
+        }
+
+        switch ("${{ inputs.target-arch }}") {
+          "x86" { $TargetArch = "x86" }
+          "amd64" { $TargetArch = "amd64" }
+          "arm64" { $TargetArch = "arm64" }
+          "" { $TargetArch = $HostArch }
+          default {
+            Write-Output "::error::Unsupported target architecture: `"${{ inputs.target-arch }}`""
+            exit 1
+          }
+        }
+
+        Write-Output "Host OS: $HostOs"
+        Write-Output "Host architecture: $HostArch"
+        Write-Output "Target architecture: $TargetArch"
+
+        $VerifiedInput = @"
+        host-os=$HostOs
+        host-arch=$HostArch
+        target-arch=$TargetArch
+        "@
+        Write-Output $VerifiedInput | Out-File -FilePath $env:GITHUB_OUTPUT -Encoding utf8 -Append
+
+    - name: Install Windows SDK version ${{ inputs.windows-sdk-version }}
+      if: steps.verify-input.outputs.host-os == 'windows' && inputs.windows-sdk-version != ''
+      shell: pwsh
+      run: |
+        $WinSdkVersionString = "${{ inputs.windows-sdk-version }}"
+        $WinSdkVersion = [System.Version]::Parse($WinSdkVersionString)
+        $WinSdkVersionBuild = $WinSdkVersion.Build
+
+        $Win10SdkRoot = Get-ItemPropertyValue `
+          -Path "Registry::HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows Kits\Installed Roots" `
+          -Name "KitsRoot10"
+        $Win10SdkLib = Join-Path $Win10SdkRoot "Lib"
+        $Win10SdkInclude = Join-Path $Win10SdkRoot "Include"
+        $Win10SdkIncludeVersion = Join-Path $Win10SdkInclude $WinSdkVersionString
+
+        if (Test-Path -Path $Win10SdkIncludeVersion -PathType Container) {
+          Write-Output "Windows SDK ${WinSdkVersionString} already installed."
+        } else {
+          # Install the missing SDK.
+          Write-Output "Installing Windows SDK ${WinSdkVersionString}..."
+
+          $InstallerLocation = Join-Path "${env:ProgramFiles(x86)}" "Microsoft Visual Studio" "Installer"
+          $VsWhere = Join-Path "${InstallerLocation}" "vswhere.exe"
+          $VsInstaller = Join-Path "${InstallerLocation}" "vs_installer.exe"
+          $InstallPath = (& "$VsWhere" -latest -products * -format json | ConvertFrom-Json).installationPath
+          $process = Start-Process "$VsInstaller" `
+              -PassThru `
+              -ArgumentList "modify", `
+                  "--installPath", "`"$InstallPath`"", `
+                  "--channelId", "https://aka.ms/vs/17/release/channel", `
+                  "--quiet", "--norestart", "--nocache", `
+                  "--add", "Microsoft.VisualStudio.Component.Windows11SDK.${WinSdkVersionBuild}"
+          $process.WaitForExit()
```

Comment:
Why the `$process.WaitForExit()` rather than have `Start-Process` use the `-Wait` option?